### DISCUSSION TITLE
Python Issues with Pikaraoke

### DISCUSSION THREAD

**Discussion by CrimsonMoose** [vicwomg/pikaraoke#309] [Q&A] [answered]
initially running a default/clean Pi4 OS but that had 3.4.x for python3 which doesn't support the walrus operator (Syntax error on "path :=") So I updated python to 3.9.17

Now pikaraoke setup fails around "Collecting Pycryptodomex" and setup never succeeds 

Manually running pip3 install pycryptodomex ends up with the same error
"Preparing metadata (pyproject.toml) did not run successfully
...90 lines of ourput...
Last line is 
ModuleNotFoundError: No module named '_ctypes'

Any one know how to set this one up properly?

**Comment by vicwomg** [comment]
Which pi os version? Bullseye is recommended for now, which comes with 3.9

**Comment by vicwomg** [comment, marked as answer]
Oops, I meant bookworm! Debian Bookworm.

**Comment by CrimsonMoose** [comment]
Thanks! the OS was the original from years ago, reimaging with new OS (32bit) had the newer python version and the install was silky smooth
